tests/Fixer/ControlStructure/SingleExpressionPerLineFixerTest.php

```diff
@@ -0,0 +1,329 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ControlStructure;
+
+use PhpCsFixer\Fixer\ControlStructure\SingleExpressionPerLineFixer;
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Vincent Langlet
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ControlStructure\SingleExpressionPerLineFixer
+ */
+final class SingleExpressionPerLineFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param array<string, mixed> $config
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null, array $config = []): void
+    {
+        $this->fixer->configure($config);
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<array{0: string, 1?: null|string, 2?: array<string, mixed>}>
+     */
+    public static function provideFixCases(): iterable
+    {
+        yield ['<?php $a = [1, 2, 3];'];
+
+        yield [
+            '<?php
+                    $a = [
+1,
+                        2
+];',
+            '<?php
+                    $a = [1,
+                        2];',
+        ];
+
+        yield [
+            '<?php
+                    $a = [
+                        1,
+2,
+                        3
+                    ];',
+            '<?php
+                    $a = [
+                        1, 2,
+                        3
+                    ];',
+        ];
+
+        yield ['<?php $a = array(1, 2, 3);'];
+
+        yield [
+            '<?php
+                    $a = array(
+                        1,
+2,
+                        3
+                    );',
+            '<?php
+                    $a = array(
+                        1, 2,
+                        3
+                    );',
+        ];
+
+        yield ['<?php $a = foo(1, 2, 3);'];
+
+        yield [
+            '<?php
+                    $a = foo(1, 2,
+                        3
+                    );',
+        ];
+
+        yield [
+            '<?php
+                    $a = foo(
+1,
+2,
+                        3
+                    );',
+            '<?php
+                    $a = foo(1, 2,
+                        3
+                    );',
+            ['elements' => [SingleExpressionPerLineFixer::ELEMENTS_ARGUMENTS]],
```

Comment:
I think it's better not to use such constants here and to use hardcoded values instead. You want your tests to fail if the accepted values change in the implementation, which is not possible when the implementation's constants are also used in the test.

Then maybe the constants could be made private in the implementation.